File: regexlib/2021-5-15/python_re_test_file/regexlib_4272.py
# 4272
# ^(((\.\.){1}/)*|(/){1})?(([a-zA-Z0-9]*)/)*([a-zA-Z0-9]*)+([.jpg]|[.gif])+$
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"f"*32+"!1 _!\n_SLQ_3"

import re
from time import perf_counter

regex = """^(((\.\.){1}/)*|(/){1})?(([a-zA-Z0-9]*)/)*([a-zA-Z0-9]*)+([.jpg]|[.gif])+$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "f" * i * 1 + "!1 _!\n_SLQ_3"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_4272.js
/* 4272
 * ^(((\.\.){1}/)*|(/){1})?(([a-zA-Z0-9]*)/)*([a-zA-Z0-9]*)+([.jpg]|[.gif])+$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"f"*32+"!1 _!\n_SLQ_3"
 */
var REGEX = new RegExp("^(((\.\.){1}/)*|(/){1})?(([a-zA-Z0-9]*)/)*([a-zA-Z0-9]*)+([.jpg]|[.gif])+$")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'f'.repeat(i*1) + '!1 _!\n_SLQ_3'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}